components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsStorage.kt

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.sitepermissions
+
+import android.content.Context
+import android.support.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import mozilla.components.feature.sitepermissions.db.SitePermissionsDatabase
+import mozilla.components.feature.sitepermissions.db.toSitePermissionsEntity
+
+/**
+ * A storage implementation to save [SitePermissions].
+ *
+ */
+class SitePermissionsStorage(
+    context: Context
+) {
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var databaseInitializer = {
+        SitePermissionsDatabase.get(context)
+    }
+
+    private val database by lazy { databaseInitializer() }
+
+    internal fun save(sitePermissions: SitePermissions) {
+        CoroutineScope(Dispatchers.IO).launch {
+            database
+                .sitePermissionsDao()
+                .insertOrUpdateSitePermission(
+                    sitePermissions.toSitePermissionsEntity()
+                )
+        }
+    }
+
+    fun update(sitePermissions: SitePermissions) {
+        save(sitePermissions)
+    }
+
+    fun findSitePermissionsBy(origin: String): Deferred<SitePermissions?> {
```

Comment:
Can I make a request for `findOriginsBySitePermission`? 😃 

I like Android's list system of permissions. You can see a single app's permissions or all the apps that have a permission. Maybe this can be something for the future, if not now.

Reply:
To do something similar to this but with sites permissions?
![screenshot_20190226-183814](https://user-images.githubusercontent.com/773158/53454771-47bfc900-39f6-11e9-9ea0-d14ddffd7c8b.png)


![screenshot_20190226-183914](https://user-images.githubusercontent.com/773158/53454847-8f465500-39f6-11e9-88fd-562e4037a5f6.png)

Reply:
Looks like that this is what chrome is doing right now, you can mange the permissions by site or by type
![screenshot_20190226-201441](https://user-images.githubusercontent.com/773158/53458257-5c568e00-3a03-11e9-91fd-a385729c22c3.png)

By specify permission.
![screenshot_20190226-201519](https://user-images.githubusercontent.com/773158/53458316-9de73900-3a03-11e9-89ac-9d0934d21dcd.png)

Reply:
@jonalmeida  I will try to include it 👍  in this version